### ISSUE TITLE
RESMLPS.forward uses .numpy() and hard-coded .cuda()

### ISSUE THREAD

**Issue by FIRS-ZOFIS** [LianpoWang/SLIIF#14]
In RESMLPS.forward:

res_list = [mlp(x, cood_out).cpu().detach().numpy() for mlp in self.mlps]
mlp_res = torch.tensor(res_list).cuda()

This breaks autograd (graph is lost after .detach().numpy()) and forces CUDA usage, making the model non-trainable on CPU or arbitrary devices.
Better to keep everything as tensors on x.device, e.g.:

res_list = [mlp(x, cood_out) for mlp in self.mlps]
mlp_res = torch.stack(res_list, dim=0).to(x.device)
